artifactory/services/repositories.go

```diff
@@ -60,6 +58,21 @@ func (rs *RepositoriesService) GetWithFilter(params RepositoriesFilterParams) (*
 	return repoDetails, errorutils.CheckError(err)
 }
 
+func createWithFilterUrl(params RepositoriesFilterParams) string {
+	endpoint := apiRepositories
+	if params.RepoType != "" {
+		endpoint += "?type=" + params.RepoType
+	}
+	if params.PackageType != "" {
+		endpoint += "&packageType=" + params.PackageType
+	}
+	if params.ProjectKey != "" {
+		log.Info("Getting repositories with project key:", params.ProjectKey)
+		endpoint += "&project=" + params.ProjectKey
+	}
```

Comment:
what happen if only package type or only project will be provided with the "?"/"&" signs?
The "?" should be always first

Reply:
i suggest checking this func with all possible combinations of this 3 params